go/adbc/driver/databricks/stmt_record_reader.go

```diff
@@ -208,6 +213,19 @@ func (r *statementReader) Next() bool {
 
 	if r.activeChunk.Next() {
 		r.rec = r.activeChunk.reader.Record()
+		// Validate schema compatibility
+		chunkSchema := r.activeChunk.Schema()
+		if len(r.schema.Fields()) != len(chunkSchema.Fields()) {
+			r.err = fmt.Errorf("chunk schema field count mismatch: expected %d, got %d", len(r.schema.Fields()), len(chunkSchema.Fields()))
+			return false
+		}
+		for i, field := range r.schema.Fields() {
+			chunkField := chunkSchema.Field(i)
+			if field.Type.ID() != chunkField.Type.ID() {
```

Comment:
We need full equality here. So we should compare the `Type.Fingerprint()`.

Reply:
Since we are moving the validation to `.Schema()` instead, the only thing we have to do here is a swap of the `r.rec.schema` for the `r.Schema()` which is enriched with our metadata. Think about the Release/Retain calls you will need for this maneuver.